### PR TITLE
58 type annotation on other nodes

### PR DESCRIPTION
**Pull request by FBroy**
# Description

Added type annotation to the 'and' and the 'or' operators. The annotations are grouped into the annotateLogicalOperators file.

## Changes

1. Add type annotation for the 'and' operator.
2. Add type annotation for the 'or' operator.

## Deletes Source Branch?

No:     Work is (partially) done, but more work will be done on the source branch.

## How Many Approvals?

* 2 Approvals (Normal features and bugs)

### CODE REVIEW

File: src/typeInference/annotateLogicalOperator.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import { SequenceMultiplicity, SequenceType, ValueType } from '../expressions/dataTypes/Value';
+import astHelper, { IAST } from '../parsing/astHelper';
+
+export function annotateLogicalOperator(ast: IAST): SequenceType {
+	switch (ast[0]) {
+		case 'andOp':
+			return annotateAndOperator(ast);
+		case 'orOp':
+			return annotateOrOperator(ast);
+	}
```
Comment (yuanzexiong) [inline src/typeInference/annotateLogicalOperator.ts:15]: I like the sub switch tree that covers the specific cases in here. The annotateAST switch cases are growing into something a bit scary already.

**Review by yuanzexiong — APPROVED** [review]
Very well done!

P.s. I added a few lines of docs as part of my docs merge request

File: src/typeInference/annotateLogicalOperator.ts (hunk 1 above)
Comment (HectorPeeters) [reply src/typeInference/annotateLogicalOperator.ts:15]: I agree that it keeps the annotateAst file cleaner but it does feel a bit redundant. Not sure if can be solved easily though

**Review by HectorPeeters — APPROVED** [review]

File: src/typeInference/annotateLogicalOperator.ts (hunk 1 above)
Comment (maanpalle) [reply src/typeInference/annotateLogicalOperator.ts:15]: We can later on always predict the type of operator by the length of the ast at that point.

**Pull request merged**